### PR TITLE
#152 - Update some images of the website

### LINKED ISSUE #152: Low DPI images
### Problem

I think all / most of the raster images in the groupincome.org website are low DPI and look blurry on high DPI displayz.

Example:

  * The images in the `faces/` directory
  * https://groupincome.org/images/about/graph1.png
  * https://groupincome.org/images/about/okturtles.png
  * https://groupincome.org/images/temp/homepage-panel7.png

Some of them are high DPI (2x), but they have no low DPI version, which means that they are wasting bandwidth on low DPI devices.

Examples:

  * https://groupincome.org/images/temp/homepage-panel8_2x.jpg

### Solution

There are two solutions

1. Generate images at various sizes and present them using the `srcset` attribute. E.g., ``. I think common sizes are 1x, 1.5x, 2x, 3x and 4x.
2. Even better, most images seem like they could be vectors (SVG) instead.

### PR DESCRIPTION
**Pull request by SebinSong**
closes #152 

### Summary of the PR
1. Replace some raster images with the SVG equivalent.
2. Minor design bug-fix in `/get-started` page and add `srcset` attribute for the raster images used in the cards.
[ From ]

[ To ]

3. Add `srcset` attribute to all face images in `/home` page for high DPI.

### CODE REVIEW

File: src/_app.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
-  app.directive('href', (el, binding) => updateAttr('href', el, binding.value))
+  app.directive('src', (el, binding) => {
+    setResolvedPathToAttr('src', el, binding.value)
+  })
+  app.directive('srcset', (el, binding) => {
+    el.setAttribute(
+      'srcset',
+      imgPathToSrcSet(binding.value, parseInt(binding.arg || '1'))
+    )
+  })
```
Comment (SebinSong) [inline src/_app.ts:18]: Within `.vue` component, we can use `v-srcset:[number]='path'` for conveniently generate the `image_1x.png 1x, image_2x.png 2x` type of string.

File: src/_app.ts [hunk 2]
```diff
@@ -1,14 +1,23 @@
 import type { App } from 'vue'
-import { resolvePath } from '@/utils/helpers.js'
+import { resolvePath, imgPathToSrcSet } from '@/utils/helpers.js'
 
-const updateAttr = (attrName, el, relPath) => {
-  el.setAttribute(attrName, resolvePath(relPath))
+const setResolvedPathToAttr = (attrName: string, el: HTMLElement, relPath: string) => {
+  el.setAttribute(attrName, resolvePath(relPath) || '')
+  console.log('!@# output path: ', resolvePath(relPath) || '')
```
Comment (taoeffect) [inline src/_app.ts]: Is this debug logging still needed?

File: src/_app.ts [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
   // custom-directives
-  app.directive('src', (el, binding) => updateAttr('src', el, binding.value))
-  app.directive('href', (el, binding) => updateAttr('href', el, binding.value))
+  app.directive('src', (el, binding) => {
+    setResolvedPathToAttr('src', el, binding.value)
+  })
+  app.directive('srcset', (el, binding) => {
+    el.setAttribute(
+      'srcset',
+      imgPathToSrcSet(binding.value, parseInt(binding.arg || '1'))
```
Comment (taoeffect) [inline src/_app.ts:16]: As far as naming conventions go, my understanding is that most people use this convention: `image.jpg` and `image@2x.jpg`.

One of the earliest libraries to automatically and conveniently load these images is [retina.js](https://imulus.github.io/retinajs/). That's the format it uses.

I asked Claude, "what is the standard method of naming image files for high dpi displays and including them in srcset?"

And this is what it said:

Screenshot 2024-10-22 at 1 21 39 PM

**Review by taoeffect — CHANGES_REQUESTED** [review]
Nice work @SebinSong! It seems to be working well enough. I left some feedback.

File: src/utils/helpers.js [hunk 4]
```diff
@@ -9,6 +9,29 @@ export function resolvePath (relPath = '') {
...
+  if (!relPath) { return undefined }
+
+  const filename = relPath.split('/').pop()
+  const [imgName, ext] = filename.split('.')
+  const folderPath = relPath.replace(filename, '')
+  let output = []
+
+  for (let i=1; i<=dpi; i++) {
+    output.push(
+      resolvePath(`${folderPath}${imgName}` + (i>1 ? `@${i}x.` : '.') + ext) +
```
Comment (taoeffect) [inline src/utils/helpers.js:27]: I'm seeing that this does use <code>i>1 ? `@${i}x.` : '.'</code> here, but the files have been renamed to use `@1x.jpg`, which is not the standard convention. The standard convention doesn't use any suffix for 1x files...

**Review by taoeffect — CHANGES_REQUESTED** [review]
Question/cocnern:

File: src/utils/helpers.js (hunk 4 above)
Comment (SebinSong) [reply src/utils/helpers.js:27]: fixed...

**Review by taoeffect — COMMENTED** [review]
Regarding the 1x images of faces... I see they've all been modified, however, they've been modified in different ways. In one, the size is decreased, and in the other, it's increased:

Screenshot 2024-10-23 at 12 01 18 PM
Screenshot 2024-10-23 at 12 01 21 PM

Is this intentional?

**Comment by SebinSong** [comment]
@taoeffect I used figma's export functionality for getting 1x, 2x .. images and that's what it gave me when I selected 1x option...

image

**Review by taoeffect — APPROVED** [review]
Fantastic work @SebinSong !

**Pull request merged**